src/client/app/components/admin/AddMetersComponent.tsx

```diff
@@ -0,0 +1,69 @@
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react';
+import Dropzone from 'react-dropzone';
+import { fileProcessingApi } from '../../utils/api';
+import { showSuccessNotification, showErrorNotification } from '../../utils/notifications';
+
+interface AddMetersProps {
+	fetchMeterDetailsIfNeeded(alwaysFetch?: boolean): Promise<any>;
+}
+export default class AddMetersComponent extends React.Component<AddMetersProps, {}> {
+	constructor(props: AddMetersProps) {
+		super(props);
+		this.handleMeterToImport = this.handleMeterToImport.bind(this);
+	}
+
+	public handleMeterToImport(files: File[]) {
+		const file = files[0];
+		let meters = [];
+		const reader = new FileReader();
+		let listValue = [];
+		reader.onload = () => {
+			const fileAsBinaryString = reader.result;
```

Comment:
I feel like all of this could be much more readable if it was replaced with a for loop that populated a new list. And `listValue` isn't a great variable name. I wouldn't know what this code was doing unless I already knew the format we want csv files to be in.

Also, this breaks on csv files that quote everything, which is a valid choice for csvs.

Reply:
changes made. now it accepts file with quotes. Also, it is much more readable.